--- a/etmem/CMakeLists.txt	2024-07-23 23:00:35.754062342 +0800
+++ b/etmem/CMakeLists.txt	2024-07-31 17:27:12.162733127 +0800
@@ -93,6 +93,12 @@ target_include_directories(etmemd_scan P
  ${PROJECT_SOURCE_DIR}/inc/etmemd_inc
  ${GLIB2_INCLUDE_DIRS})
 
+# remove -D_FORTIFY_SOURCE=1 from CMAKE_CFLAGS set by buildroot
+separate_arguments(C_FLAGS_LIST UNIX_COMMAND "${CMAKE_C_FLAGS}")
+list(REMOVE_ITEM C_FLAGS_LIST "-D_FORTIFY_SOURCE=1")
+string(REPLACE ";" " " NEW_C_FLAGS "${C_FLAGS_LIST}")
+set(CMAKE_C_FLAGS "${NEW_C_FLAGS}" CACHE STRING "Modified C flags" FORCE)
+
 target_compile_options(etmemd PRIVATE -fsigned-char -fno-omit-frame-pointer -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wmissing-declarations -fno-strict-aliasing -Werror -Wformat -Wformat-security -D_GNU_SOURCE -fPIE -pie -fPIC -fstack-protector-strong -fno-common -DNDEBUG -O2 -D_FORTIFY_SOURCE=2 -Wall -Werror -Wl,-z,relro,-z,now,-z,noexecstack -Wtrampolines -pthread -Wno-pointer-sign -Wstrict-prototypes -Wold-style-definition -std=gnu99)
 
 
